cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -315,10 +316,30 @@ public ApplicationDetail getAppDetail(ApplicationId appId) throws Exception {
     if (appSpec == null) {
       throw new ApplicationNotFoundException(appId);
     }
+    // TODO: fetch creation time from store
     String ownerPrincipal = ownerAdmin.getOwnerPrincipal(appId);
     return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appSpec, ownerPrincipal));
   }
 
+  /**
+   * Get details about the latest version of the specified application
+   *
+   * @param appId the id of the application to get
+   * @return detail about the latest version of the specified application
+   * @throws ApplicationNotFoundException if the specified application does not exist
+   */
+  public ApplicationDetail getAppLatestVersionDetail(ApplicationId appId) throws Exception {
+    // TODO: CDAP-12473: filter based on the entity visibility in the app detail
```

Comment:
I don't understand this comment, what needs to be changed here?

Reply:
This comment is an error - was probably copied over from another place by mistake - will remove